### ISSUE TITLE
Please allow for registering custom PolledConfigurationSource with Archaius

### ISSUE THREAD

**Issue by digitaljoel** [spring-cloud/spring-cloud-netflix#218]
We have a custom PolledConfigurationSource to get our custom configuration system hooked in to Archaius.  With the ArchaiusAutoConfiguration I don't see a way to add our configuration source to the configuration setup in there.

**Comment by dsyer** [comment]
Archaius is fairly primitive (lots of static singletons) so you can probably do it however you would in a native netflix app? Did you try something obvious and it failed? More details would help.

**Comment by digitaljoel** [comment]
Sorry for the brevity in the initial submission.

In my application I attempted to register our configuration source with ConfigurationManager.install  before calling SpringApplication.run (within my main method).  On doing that it complained that when it attempted to do the ArchaiusAutoConfiguration that a configuration is already registered and the app failed to start.  I was hoping that there would a be a way to tell the auto configuration to also add in some custom list of configurations via injecting beans of the appropriate type and registering them if any are configured, or something like that.

Another option would be to simply check for an already-registered config in the ArchaiusAutoConfiguration and then conditionally install or append, such as:

      if ( ConfigurationManager.isConfigurationInstalled()) {
        ConfigurationManager.getConfigInstance().append(config);
      }
      else {
        ConfigurationManager.install(config);
      }

That would allow anyone to register their configuration as needed, then still use the auto configuration provided in spring-cloud-netflix-core.

**Comment by icemanftg** [comment]
I agree with @digitaljoel on the check for ArchaiusAutoConfiguration part. Otherwise someone has to manually append the configuration after the spring context has been loaded.

**Comment by dsyer** [comment]
Contributions would be more than welcome.

**Comment by icemanftg** [comment]
I've just tried the approach above and it kinda solves the problem, the db properties are pooled by archaius, on the other hand, if there is another system source before the ArchaiusAutoConfiguration (with this modification) finishes, the properties from the config.properties file are not pooled anymore. 
```
// below come from ConfigurationManager.createDefaultConfigInstance()
DynamicURLConfiguration defaultURLConfig = new DynamicURLConfiguration();
try {
	config.addConfiguration(defaultURLConfig, URL_CONFIG_NAME);
}
catch (Throwable ex) {
	log.error("Cannot create config from " + defaultURLConfig, ex);
}
```
I guess it has something to do with the archaius hierarchy ![Archaius hierarchy](http://3.bp.blogspot.com/-LvvCo-wWjlo/T9-QCxI60KI/AAAAAAAAAAg/CMOsvfuhn5k/s1600/Archaius+Composite+Configuration+Diagram.png)

I'll look more into it to see if I can find a plausible solution.

**Comment by spencergibb** [comment]
closed via 611c0b5c1c425f29e6492744e88fc4fe79d81e4d
